torch/_functorch/_aot_autograd/input_output_analysis.py

```diff
@@ -418,15 +341,11 @@ def compute_overlapping_inputs(aot_config, fwd_inputs, aliased_input_indices):
             suppress_guards = shape_env.suppress_guards
 
     with suppress_guards():
-        for j in range(num_aliases):
-            for i in range(j):
-                j_ = aliased_input_indices[j]
-                i_ = aliased_input_indices[i]
-                if not _tensors_definitely_do_not_overlap(
-                    fwd_inputs[i_], fwd_inputs[j_]
-                ):
-                    actual_aliased_indices.add(i_)
-                    actual_aliased_indices.add(j_)
+        aliased_fwd_inputs = [fwd_inputs[i] for i in aliased_input_indices]
+        actual_aliased_indices = {
+            aliased_input_indices[i]
+            for i in compute_overlapping_tensors(aliased_fwd_inputs)
+        }
```

Comment:
Just noticed in the below code: if there are no dynamic shapes involved (the shapes are all static), we don't actually need `StorageOverlap` guards (since our `TENSOR_MATCH` guards already guard on the exact size/stride/offset). Are we making sure to only generate these new guards if any of the tensors' shapes in question are dynamic?

Reply:
Ah, good catch. I will guard for that, too.